### PR TITLE
Add documentation for ide0130

### LINKED ISSUE #65759: Add documentation for IDE0130
<html>
<body>

IDE0130 | https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0130 | Namespace does not match folder structure
-- | -- | --

</body>
</html>

Steps to follow to add documentation: https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md

### PR DESCRIPTION
**Pull request by ryzngard**
## Summary

Add documentation for ide0130 (match namespace and folder structure).

Fixes https://github.com/dotnet/roslyn/issues/65759

### CODE REVIEW

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- VB
+---
+# Namespace does not match folder structure (IDE0130)
+
+| Property                 | Value                                                  |
+| ------------------------ | ------------------------------------------------------ |
+| **Rule ID**              | IDE0130                                                |
+| **Title**                | Namespace does not match folder structure              |
+| **Category**             | Style                                                  |
+| **Subcategory**          | Naming  rules                                          |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:23]:
```suggestion
| **Subcategory**          | Language rules                                        |
```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+| Property                 | Value                                                | Description                                        |
+| ------------------------ | ---------------------------------------------------- | -------------------------------------------------- |
+| **Option name**          | dotnet_style_namespace_match_folder                  |                                                    |
+| **Option values**        | `true`                                               | Prefer namespace naming to match folder structure. |
+|                          | `false`                                              | Disables the rule.                                 |
+| **Default option value** | `false`                                              |                                                    |
+
+## Example
+
+Assume that the following file is `Data\Example.cs`, where `Data` represents the folder structure from the project file. Also note that the structure is added to the root namespace, which in this example will be `Root`.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:46]:
```suggestion
Assume that the following code snippets are from a file named `Data/Example.cs`, where `Data` represents the folder structure from the project file. The folder structure naming is added to the root namespace, which in this example is `Root`.
```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 1 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:23]: I don't think this is a "naming rule" in terms of how "naming rules" are used in the docs.

**Review by gewarren — COMMENTED** [review]
Thanks for adding this. Can you also add to the TOC at https://github.com/dotnet/docs/blob/446d1407d6cf814b11b06bd0b203d5245b9dd69c/docs/fundamentals/toc.yml#L1679

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 1 above)
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:23]: Agree. I had this change and forgot to hit save :/

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+description: "Learn about code analysis rule IDE0130: Namespace does not match folder structure"
+ms.date: 12/05/2021
+ms.topic: reference
+f1_keywords:
+- IDE0130
+helpviewer_keywords:
+- IDE0130
+author: gewarren
+ms.author: gewarren
+dev_langs:
```
[the remark this answers is not in the document]
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:12]: Will do

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+---
+title: "IDE0130: Namespace does not match folder structure"
+description: "Learn about code analysis rule IDE0130: Namespace does not match folder structure"
+ms.date: 12/05/2022
+ms.topic: reference
+f1_keywords:
+- IDE0130
+helpviewer_keywords:
+- IDE0130
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:9]: @gewarren Do we add the code-style option in f1_keywords and helpviewer_keywords?

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+Options specify the behavior that you want the rule to enforce. For information about configuring options, see [Option format](language-rules.md#option-format).
+
+### dotnet_style_namespace_match_folder
+
+| Property                 | Value                                                | Description                                        |
+| ------------------------ | ---------------------------------------------------- | -------------------------------------------------- |
+| **Option name**          | dotnet_style_namespace_match_folder                  |                                                    |
+| **Option values**        | `true`                                               | Prefer namespace naming to match folder structure. |
+|                          | `false`                                              | Disables the rule.                                 |
+| **Default option value** | `false`                                              |                                                    |
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:42]: Isn't the default value `true`?

https://github.com/dotnet/roslyn/blob/19704e6876c4a5a88267af5cf83a4bfe6287d356/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/IdeCodeStyleOptions.cs#L64

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+## Example
+
+Assume that the following code snippets are from a file named `Data/Example.cs`, where `Data` represents the folder structure from the project file. The folder structure naming is added to the root namespace, which in this example is `Root`.
+
+### C\#
+
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:49]:
```suggestion
```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    {
+        public void M()
+        {
+        }
+    }
+}
+```
+
+### VB
+
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:75]:
```suggestion
```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+- IDE0130
+helpviewer_keywords:
+- IDE0130
+author: gewarren
+ms.author: gewarren
+dev_langs:
+- CSharp
+- VB
+---
+# Namespace does not match folder structure (IDE0130)
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:16]: IIRC, there was some requirement about a specific MSBuild property being set or something like that, can't recall exactly.

If that is still true, can you document it?

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+| Property                 | Value                                                | Description                                        |
+| ------------------------ | ---------------------------------------------------- | -------------------------------------------------- |
+| **Option name**          | dotnet_style_namespace_match_folder                  |                                                    |
+| **Option values**        | `true`                                               | Prefer namespace naming to match folder structure. |
+|                          | `false`                                              | Disables the rule.                                 |
+| **Default option value** | `false`                                              |                                                    |
+
+## Example
+
+Assume that the following code snippets are from a file named `Data/Example.cs`, where `Data` represents the folder structure from the project file. The folder structure naming is added to the root namespace, which in this example is `Root`.
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:46]:
```suggestion
Assume that the following code snippets are from a file named `Data/Example.cs` or `Data/Example.vb`, where `Data` represents the folder structure from the project file. The folder structure naming is added to the root namespace, which in this example is `Root`.
```

**Review by Youssef1313 — COMMENTED** [review]
I think this page needs an update as well:

https://github.com/dotnet/docs/blob/main/docs/fundamentals/code-analysis/style-rules/index.md

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 8 above)
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:16]: It's only _sort_ of true. That's where the nuance in running with a workspace vs not comes in. To get in command line builds you have to set the project root, but for VS it is provided through the workspace context (via project system). 

I wasn't sure how much nuance we wanted to add to this page

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 5 above)
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:42]: Oh yes! I forgot to update here. It's true with severity suggestion

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 6 above)
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:49]: Do we not want headers here? Just a styling question, I'm happy either way

Comment (Youssef1313) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:49]: @ryzngard The site will show a selector in the page and show only one of the snippets if they are put one after another without headers in between. For example:

image

This is what `dev_langs` is for.

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 8 above)
Comment (Youssef1313) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:16]: Running analyzers in command-line builds is a core scenario IMO.

If the user should do something special, it's better to be documented.

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 5 above)
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:42]: Do we not supply the default severity somewhere?

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 4 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:9]: It's probably a good idea.

File: docs/fundamentals/code-analysis/style-rules/ide0130.md (hunk 8 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/style-rules/ide0130.md:16]: Yes, I agree with @Youssef1313 .

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+| Property                 | Value                                                | Description                                        |
+| ------------------------ | ---------------------------------------------------- | -------------------------------------------------- |
+| **Option name**          | dotnet_style_namespace_match_folder                  |                                                    |
+| **Option values**        | `true`                                               | Prefer namespace naming to match folder structure. |
+|                          | `false`                                              | Disables the rule.                                 |
+| **Default option value** | `true`                                               |                                                    |
+
+> [!NOTE]
+> `dotnet_style_namespace_match_folder` depends on knowing the current project and root namespace properties. This is provided by Visual Studio but not done in command line builds. For command line builds, such as `dotnet build` to work, you must add the following properties to your `.editorconfig`.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:45]:
```suggestion
> The `dotnet_style_namespace_match_folder` option depends on knowing the current project and root namespace properties. This information is provided by Visual Studio but is not available for command-line builds, such as `dotnet build`. For command-line builds to work, you must add the following properties to your *.editorconfig* file:

```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
+---
+title: "IDE0130: Namespace does not match folder structure"
+description: "Learn about code analysis rule IDE0130: Namespace does not match folder structure"
+ms.date: 12/05/2022
+ms.topic: reference
+f1_keywords:
+- IDE0130
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:7]:
```suggestion
- IDE0130
- dotnet_style_namespace_match_folder
```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+helpviewer_keywords:
+- IDE0130
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:9]:
```suggestion
- IDE0130
- dotnet_style_namespace_match_folder
```

File: docs/fundamentals/code-analysis/style-rules/ide0130.md [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+| ------------------------ | ---------------------------------------------------- | -------------------------------------------------- |
+| **Option name**          | dotnet_style_namespace_match_folder                  |                                                    |
+| **Option values**        | `true`                                               | Prefer namespace naming to match folder structure. |
+|                          | `false`                                              | Disables the rule.                                 |
+| **Default option value** | `true`                                               |                                                    |
+
+> [!NOTE]
+> The `dotnet_style_namespace_match_folder` option depends on knowing the current project and root namespace properties. This information is provided by Visual Studio but is not available for command-line builds, such as `dotnet build`. For command-line builds to work, you must add the following properties to your *.editorconfig* file and replace the root directory and namespace placeholders with your values:
+
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:48]: >

Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0130.md:48]:
```suggestion
>
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**